[2025-04-25T11:52:40.571+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-25T11:52:39.290744+00:00 [queued]>
[2025-04-25T11:52:40.587+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-25T11:52:39.290744+00:00 [queued]>
[2025-04-25T11:52:40.589+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-04-25T11:52:40.611+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): submit_traffic_processor_spark_job> on 2025-04-25 11:52:39.290744+00:00
[2025-04-25T11:52:40.622+0000] {standard_task_runner.py:60} INFO - Started process 183 to run task
[2025-04-25T11:52:40.631+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'traffic_data_pipeline', 'submit_traffic_processor_spark_job', 'manual__2025-04-25T11:52:39.290744+00:00', '--job-id', '16', '--raw', '--subdir', 'DAGS_FOLDER/traffic_pipeline_dag.py', '--cfg-path', '/tmp/tmp3z4vtzwn']
[2025-04-25T11:52:40.634+0000] {standard_task_runner.py:88} INFO - Job 16: Subtask submit_traffic_processor_spark_job
[2025-04-25T11:52:40.741+0000] {task_command.py:423} INFO - Running <TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-25T11:52:39.290744+00:00 [running]> on host 237827c6637e
[2025-04-25T11:52:41.058+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='traffic_data_pipeline' AIRFLOW_CTX_TASK_ID='submit_traffic_processor_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-04-25T11:52:39.290744+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-25T11:52:39.290744+00:00'
[2025-04-25T11:52:41.060+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-25T11:52:41.062+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n            export KAFKA_BROKERS_INTERNAL="kafk-1:9092,kafk-2:9092,kafk-3:9092" && \\\n            export KAFKA_TOPIC="raw_traffic_data" && \\\n            export SCHEMA_REGISTRY_URL="http://schema-registry:8081" && \\\n            export REDIS_HOST="redis" && \\\n            export CLICKHOUSE_HOST="clickhouse-server" && \\\n            export CLICKHOUSE_NATIVE_PORT="9000" && \\\n            export MINIO_ENDPOINT="http://minio:9000" && \\\n            export MINIO_ACCESS_KEY="minioadmin" && \\\n            export MINIO_SECRET_KEY="minioadmin" && \\\n            export MINIO_BUCKET="traffic-data" && \\\n            export CHECKPOINT_LOCATION="s3a://traffic-data/checkpoints_***" && \\\n            export PYSPARK_PYTHON=/usr/bin/python3 && \\\n            spark-submit \\\n            --master local[*] \\\n            --deploy-mode client \\\n            --conf spark.sql.streaming.schemaInference=true \\\n            --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 \\\n            --conf spark.hadoop.fs.s3a.access.key=minioadmin \\\n            --conf spark.hadoop.fs.s3a.secret.key=minioadmin \\\n            --conf spark.hadoop.fs.s3a.path.style.access=true \\\n            --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem \\\n            --conf spark.sql.avro.schemaRegistryUrl=http://schema-registry:8081 \\\n            /opt/spark/app/processor.py\n            ']
[2025-04-25T11:52:41.074+0000] {subprocess.py:86} INFO - Output:
[2025-04-25T11:52:41.090+0000] {subprocess.py:93} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-04-25T11:52:44.506+0000] {subprocess.py:93} INFO - --- Starting Spark Streaming Traffic Processor ---
[2025-04-25T11:52:44.507+0000] {subprocess.py:93} INFO - Reading from Kafka: kafk-1:9092,kafk-2:9092,kafk-3:9092 / Topic: raw_traffic_data
[2025-04-25T11:52:44.508+0000] {subprocess.py:93} INFO - Schema Registry URL: http://schema-registry:8081
[2025-04-25T11:52:44.508+0000] {subprocess.py:93} INFO - Writing state to Redis: redis:6379
[2025-04-25T11:52:44.509+0000] {subprocess.py:93} INFO - Writing history to ClickHouse: clickhouse-server:9000 (DB: traffic_db)
[2025-04-25T11:52:44.509+0000] {subprocess.py:93} INFO - Writing Parquet to MinIO Bucket: traffic-data
[2025-04-25T11:52:44.510+0000] {subprocess.py:93} INFO - Checkpoint Location Base: s3a://traffic-data/checkpoints_***
[2025-04-25T11:52:44.615+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SparkContext: Running Spark version 3.3.4
[2025-04-25T11:52:44.682+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-04-25T11:52:44.794+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO ResourceUtils: ==============================================================
[2025-04-25T11:52:44.795+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-04-25T11:52:44.796+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO ResourceUtils: ==============================================================
[2025-04-25T11:52:44.797+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SparkContext: Submitted application: TrafficProcessor
[2025-04-25T11:52:44.823+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-04-25T11:52:44.833+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO ResourceProfile: Limiting resource is cpu
[2025-04-25T11:52:44.834+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-04-25T11:52:44.905+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SecurityManager: Changing view acls to: ***
[2025-04-25T11:52:44.906+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SecurityManager: Changing modify acls to: ***
[2025-04-25T11:52:44.907+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SecurityManager: Changing view acls groups to:
[2025-04-25T11:52:44.907+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SecurityManager: Changing modify acls groups to:
[2025-04-25T11:52:44.908+0000] {subprocess.py:93} INFO - 25/04/25 11:52:44 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2025-04-25T11:52:45.161+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO Utils: Successfully started service 'sparkDriver' on port 45429.
[2025-04-25T11:52:45.382+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO SparkEnv: Registering MapOutputTracker
[2025-04-25T11:52:45.438+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO SparkEnv: Registering BlockManagerMaster
[2025-04-25T11:52:45.457+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-04-25T11:52:45.458+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-04-25T11:52:45.462+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-04-25T11:52:45.491+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-62dce530-b60d-48bf-be2b-679c7f3affd6
[2025-04-25T11:52:45.506+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-04-25T11:52:45.521+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-04-25T11:52:45.726+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-04-25T11:52:45.885+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO Executor: Starting executor ID driver on host 237827c6637e
[2025-04-25T11:52:45.891+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-04-25T11:52:45.908+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 44719.
[2025-04-25T11:52:45.909+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO NettyBlockTransferService: Server created on 237827c6637e:44719
[2025-04-25T11:52:45.910+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-04-25T11:52:45.917+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 237827c6637e, 44719, None)
[2025-04-25T11:52:45.921+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManagerMasterEndpoint: Registering block manager 237827c6637e:44719 with 434.4 MiB RAM, BlockManagerId(driver, 237827c6637e, 44719, None)
[2025-04-25T11:52:45.923+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 237827c6637e, 44719, None)
[2025-04-25T11:52:45.924+0000] {subprocess.py:93} INFO - 25/04/25 11:52:45 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 237827c6637e, 44719, None)
[2025-04-25T11:52:46.307+0000] {subprocess.py:93} INFO - Spark Session created. Version: 3.3.4
[2025-04-25T11:52:46.308+0000] {subprocess.py:93} INFO - Broadcasting segments data...
[2025-04-25T11:52:46.463+0000] {subprocess.py:93} INFO - Segments data broadcasted.
[2025-04-25T11:52:49.200+0000] {subprocess.py:93} INFO - Reading Avro schema from file: /opt/spark/schemas/raw_traffic_event.avsc
[2025-04-25T11:52:49.204+0000] {subprocess.py:93} INFO - Successfully read Avro schema string.
[2025-04-25T11:52:49.205+0000] {subprocess.py:93} INFO - Deserializing Kafka messages using schema string from file...
[2025-04-25T11:52:49.319+0000] {subprocess.py:93} INFO - 25/04/25 11:52:49 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:49.746+0000] {subprocess.py:93} INFO - Using checkpoint location: s3a://traffic-data/checkpoints_***/traffic_processing_pipeline
[2025-04-25T11:52:49.969+0000] {subprocess.py:93} INFO - 25/04/25 11:52:49 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-04-25T11:52:50.837+0000] {subprocess.py:93} INFO - 25/04/25 11:52:50 WARN ResolveWriteToStream: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
[2025-04-25T11:52:51.104+0000] {subprocess.py:93} INFO - Streaming query 'traffic_processing_pipeline' started.
[2025-04-25T11:52:53.485+0000] {subprocess.py:93} INFO - 25/04/25 11:52:53 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:53.581+0000] {subprocess.py:93} INFO - 25/04/25 11:52:53 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.148+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.167+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.170+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.172+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.837+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.838+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.850+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:54.851+0000] {subprocess.py:93} INFO - 25/04/25 11:52:54 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:55.390+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 WARN KafkaDataConsumer: KafkaDataConsumer is not running in UninterruptibleThread. It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894
[2025-04-25T11:52:55.391+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 WARN KafkaDataConsumer: KafkaDataConsumer is not running in UninterruptibleThread. It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894
[2025-04-25T11:52:55.632+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:55.657+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR PythonUDFRunner: Python worker exited unexpectedly (crashed)
[2025-04-25T11:52:55.658+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-25T11:52:55.659+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 666, in main
[2025-04-25T11:52:55.659+0000] {subprocess.py:93} INFO -     eval_type = read_int(infile)
[2025-04-25T11:52:55.660+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^
[2025-04-25T11:52:55.660+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 595, in read_int
[2025-04-25T11:52:55.660+0000] {subprocess.py:93} INFO -     raise EOFError
[2025-04-25T11:52:55.661+0000] {subprocess.py:93} INFO - EOFError
[2025-04-25T11:52:55.661+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:55.661+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-25T11:52:55.662+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-25T11:52:55.663+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-25T11:52:55.663+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-25T11:52:55.664+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-25T11:52:55.664+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-25T11:52:55.665+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.665+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.666+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-25T11:52:55.666+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.666+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.667+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.667+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.668+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.668+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.669+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.669+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.669+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.670+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.670+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.671+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.671+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.672+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.672+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.673+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.673+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.674+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.675+0000] {subprocess.py:93} INFO - 	... 17 more
[2025-04-25T11:52:55.676+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.676+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.677+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.677+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.678+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.678+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.679+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.680+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.681+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.682+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.683+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.683+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.685+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR PythonUDFRunner: This may have been caused by a prior exception:
[2025-04-25T11:52:55.686+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.686+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.687+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.687+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.688+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.688+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.688+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.689+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.689+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.690+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.690+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.691+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.691+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.692+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.692+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.693+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.693+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.694+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.694+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.694+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.695+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.695+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.696+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.696+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.697+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.698+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.698+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.699+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.699+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.700+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.701+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.701+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.702+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR PythonUDFRunner: Python worker exited unexpectedly (crashed)
[2025-04-25T11:52:55.702+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-25T11:52:55.703+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 666, in main
[2025-04-25T11:52:55.703+0000] {subprocess.py:93} INFO -     eval_type = read_int(infile)
[2025-04-25T11:52:55.704+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^
[2025-04-25T11:52:55.704+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 595, in read_int
[2025-04-25T11:52:55.705+0000] {subprocess.py:93} INFO -     raise EOFError
[2025-04-25T11:52:55.705+0000] {subprocess.py:93} INFO - EOFError
[2025-04-25T11:52:55.706+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:55.706+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-25T11:52:55.707+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-25T11:52:55.707+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-25T11:52:55.708+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-25T11:52:55.709+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-25T11:52:55.709+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-25T11:52:55.710+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.710+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.711+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage3.processNext(Unknown Source)
[2025-04-25T11:52:55.711+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.712+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.713+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.714+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.hashAgg_doAggregateWithKeys_0$(Unknown Source)
[2025-04-25T11:52:55.714+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.processNext(Unknown Source)
[2025-04-25T11:52:55.715+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.715+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.716+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.716+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
[2025-04-25T11:52:55.717+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
[2025-04-25T11:52:55.718+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
[2025-04-25T11:52:55.718+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
[2025-04-25T11:52:55.719+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:136)
[2025-04-25T11:52:55.720+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
[2025-04-25T11:52:55.720+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
[2025-04-25T11:52:55.721+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
[2025-04-25T11:52:55.721+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-04-25T11:52:55.722+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-04-25T11:52:55.722+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2025-04-25T11:52:55.723+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.723+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.724+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.724+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.725+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.725+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.726+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.726+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.727+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.728+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.729+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.730+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.730+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.730+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.731+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.731+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.732+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.732+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.732+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.733+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.733+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.734+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.734+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.734+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.735+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.735+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.736+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.736+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.737+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.737+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.738+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.738+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.738+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR PythonUDFRunner: This may have been caused by a prior exception:
[2025-04-25T11:52:55.739+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.739+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.739+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.740+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.740+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.741+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.741+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.742+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.742+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.743+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.743+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.744+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.744+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.745+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.745+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.745+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.746+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.746+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.747+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.747+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.748+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.748+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.749+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.749+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.750+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.750+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.751+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.751+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.752+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.752+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.752+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.753+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.753+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR Executor: Exception in task 1.0 in stage 0.0 (TID 1)
[2025-04-25T11:52:55.754+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.754+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.754+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.755+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.755+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.756+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.756+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.756+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.756+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.757+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.757+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.758+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.758+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.759+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.759+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.759+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.760+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.760+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.761+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.761+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.761+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.762+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.762+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.762+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.763+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.763+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.764+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.764+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.764+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.765+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.765+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.766+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.766+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 WARN TaskSetManager: Lost task 1.0 in stage 0.0 (TID 1) (237827c6637e executor driver): org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.766+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.767+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.767+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.767+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.768+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.768+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.768+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.769+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.769+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.770+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.770+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.771+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.771+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.771+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.772+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.772+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.773+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.773+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.774+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.774+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.775+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.775+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.776+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.776+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.777+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.777+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.778+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.779+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.779+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.780+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.780+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.781+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:55.781+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR TaskSetManager: Task 1 in stage 0.0 failed 1 times; aborting job
[2025-04-25T11:52:55.856+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 ERROR MicroBatchExecution: Query traffic_processing_pipeline [id = 5e03d62e-cf04-4af4-95a7-b2203ba8cf76, runId = af9b9841-834b-401f-ba3a-3a2735d84dc1] terminated with error
[2025-04-25T11:52:55.857+0000] {subprocess.py:93} INFO - py4j.Py4JException: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
[2025-04-25T11:52:55.858+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/clientserver.py", line 617, in _call_proxy
[2025-04-25T11:52:55.859+0000] {subprocess.py:93} INFO -     return_value = getattr(self.pool[obj_id], method)(*params)
[2025-04-25T11:52:55.859+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:55.860+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 276, in call
[2025-04-25T11:52:55.860+0000] {subprocess.py:93} INFO -     raise e
[2025-04-25T11:52:55.861+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 273, in call
[2025-04-25T11:52:55.862+0000] {subprocess.py:93} INFO -     self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
[2025-04-25T11:52:55.862+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 292, in <lambda>
[2025-04-25T11:52:55.863+0000] {subprocess.py:93} INFO -     write_to_redis(df, epoch_id),
[2025-04-25T11:52:55.863+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:55.864+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 64, in write_to_redis
[2025-04-25T11:52:55.864+0000] {subprocess.py:93} INFO -     if df.isEmpty(): return
[2025-04-25T11:52:55.865+0000] {subprocess.py:93} INFO -        ^^^^^^^^^^^^
[2025-04-25T11:52:55.865+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 553, in isEmpty
[2025-04-25T11:52:55.866+0000] {subprocess.py:93} INFO -     return self._jdf.isEmpty()
[2025-04-25T11:52:55.866+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:55.867+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-25T11:52:55.867+0000] {subprocess.py:93} INFO -     return_value = get_return_value(
[2025-04-25T11:52:55.868+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:55.868+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 190, in deco
[2025-04-25T11:52:55.869+0000] {subprocess.py:93} INFO -     return f(*a, **kw)
[2025-04-25T11:52:55.869+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^
[2025-04-25T11:52:55.870+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-04-25T11:52:55.870+0000] {subprocess.py:93} INFO -     raise Py4JJavaError(
[2025-04-25T11:52:55.870+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o181.isEmpty.
[2025-04-25T11:52:55.871+0000] {subprocess.py:93} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 0.0 failed 1 times, most recent failure: Lost task 1.0 in stage 0.0 (TID 1) (237827c6637e executor driver): org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.871+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.871+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.872+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.872+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.873+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.873+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.874+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.874+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.875+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.875+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.876+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.876+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.877+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.877+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.877+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.878+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.878+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.878+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.879+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.879+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.880+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.880+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.881+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.881+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.881+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.882+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.882+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.883+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.884+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.884+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.884+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.885+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:55.886+0000] {subprocess.py:93} INFO - Driver stacktrace:
[2025-04-25T11:52:55.887+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2668)
[2025-04-25T11:52:55.887+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2604)
[2025-04-25T11:52:55.888+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2603)
[2025-04-25T11:52:55.889+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-04-25T11:52:55.889+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-04-25T11:52:55.890+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-04-25T11:52:55.891+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2603)
[2025-04-25T11:52:55.892+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1178)
[2025-04-25T11:52:55.892+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1178)
[2025-04-25T11:52:55.893+0000] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2025-04-25T11:52:55.893+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1178)
[2025-04-25T11:52:55.894+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2856)
[2025-04-25T11:52:55.894+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2798)
[2025-04-25T11:52:55.895+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2787)
[2025-04-25T11:52:55.895+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2025-04-25T11:52:55.896+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:952)
[2025-04-25T11:52:55.897+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2238)
[2025-04-25T11:52:55.897+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2259)
[2025-04-25T11:52:55.898+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2278)
[2025-04-25T11:52:55.898+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:506)
[2025-04-25T11:52:55.899+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:459)
[2025-04-25T11:52:55.899+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1(Dataset.scala:604)
[2025-04-25T11:52:55.900+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1$adapted(Dataset.scala:603)
[2025-04-25T11:52:55.900+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:3898)
[2025-04-25T11:52:55.900+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:510)
[2025-04-25T11:52:55.901+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3896)
[2025-04-25T11:52:55.901+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:52:55.901+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:52:55.902+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:52:55.902+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:55.902+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:52:55.903+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3896)
[2025-04-25T11:52:55.903+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.isEmpty(Dataset.scala:603)
[2025-04-25T11:52:55.904+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-04-25T11:52:55.904+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-04-25T11:52:55.905+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-04-25T11:52:55.905+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-04-25T11:52:55.905+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-04-25T11:52:55.906+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-04-25T11:52:55.906+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-04-25T11:52:55.906+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-04-25T11:52:55.907+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-04-25T11:52:55.907+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.sendCommand(ClientServerConnection.java:244)
[2025-04-25T11:52:55.908+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:384)
[2025-04-25T11:52:55.908+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:356)
[2025-04-25T11:52:55.909+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:106)
[2025-04-25T11:52:55.909+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-25T11:52:55.910+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-25T11:52:55.910+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-25T11:52:55.911+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-25T11:52:55.911+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-25T11:52:55.912+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:52:55.912+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:52:55.913+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:52:55.913+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:55.913+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:52:55.914+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-25T11:52:55.914+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:52:55.915+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:52:55.915+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:52:55.916+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-25T11:52:55.916+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-25T11:52:55.916+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:52:55.917+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:52:55.917+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:52:55.917+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:52:55.918+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-25T11:52:55.918+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-25T11:52:55.918+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-25T11:52:55.919+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-25T11:52:55.919+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:52:55.919+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:55.920+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-25T11:52:55.920+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-25T11:52:55.920+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:55.921+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:55.921+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:55.921+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:55.922+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:55.922+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.922+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:55.923+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.923+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.923+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:55.924+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:55.924+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:55.925+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:55.925+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:55.925+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:55.926+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:55.926+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:55.926+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:55.927+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:55.927+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:55.927+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:55.928+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:55.928+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:55.929+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:55.929+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.930+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:55.930+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:55.931+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:55.931+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:55.932+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:55.932+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:55.933+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:55.933+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:55.934+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:55.934+0000] {subprocess.py:93} INFO - 	at py4j.Protocol.getReturnValue(Protocol.java:476)
[2025-04-25T11:52:55.935+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:108)
[2025-04-25T11:52:55.935+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-25T11:52:55.935+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-25T11:52:55.936+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-25T11:52:55.936+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-25T11:52:55.936+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-25T11:52:55.937+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:52:55.937+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:52:55.937+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:52:55.938+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:55.938+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:52:55.939+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-25T11:52:55.939+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:52:55.939+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:52:55.940+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:52:55.940+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-25T11:52:55.941+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-25T11:52:55.941+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:52:55.941+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:52:55.942+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:52:55.942+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:52:55.942+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-25T11:52:55.943+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-25T11:52:55.943+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-25T11:52:55.944+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-25T11:52:55.944+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:52:55.944+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:55.944+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-25T11:52:55.945+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-25T11:52:55.987+0000] {subprocess.py:93} INFO - 25/04/25 11:52:55 WARN Schema: Ignored the com.traffic.events.RawTrafficEvent.timestamp.logicalType property ("iso8601-timestamp-millis"). It should probably be nested inside the "type" for the field.
[2025-04-25T11:52:56.006+0000] {subprocess.py:93} INFO - 25/04/25 11:52:56 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (237827c6637e executor driver): TaskKilled (Stage cancelled)
[2025-04-25T11:52:56.109+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-25T11:52:56.110+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 306, in <module>
[2025-04-25T11:52:56.134+0000] {subprocess.py:93} INFO -     main()
[2025-04-25T11:52:56.137+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 299, in main
[2025-04-25T11:52:56.150+0000] {subprocess.py:93} INFO -     streaming_query.awaitTermination()
[2025-04-25T11:52:56.152+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/streaming.py", line 107, in awaitTermination
[2025-04-25T11:52:56.153+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-25T11:52:56.154+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 196, in deco
[2025-04-25T11:52:56.169+0000] {subprocess.py:93} INFO - pyspark.sql.utils.StreamingQueryException: Query traffic_processing_pipeline [id = 5e03d62e-cf04-4af4-95a7-b2203ba8cf76, runId = af9b9841-834b-401f-ba3a-3a2735d84dc1] terminated with exception: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
[2025-04-25T11:52:56.170+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/clientserver.py", line 617, in _call_proxy
[2025-04-25T11:52:56.172+0000] {subprocess.py:93} INFO -     return_value = getattr(self.pool[obj_id], method)(*params)
[2025-04-25T11:52:56.173+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:56.174+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 276, in call
[2025-04-25T11:52:56.175+0000] {subprocess.py:93} INFO -     raise e
[2025-04-25T11:52:56.175+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 273, in call
[2025-04-25T11:52:56.176+0000] {subprocess.py:93} INFO -     self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
[2025-04-25T11:52:56.178+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 292, in <lambda>
[2025-04-25T11:52:56.179+0000] {subprocess.py:93} INFO -     write_to_redis(df, epoch_id),
[2025-04-25T11:52:56.180+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:56.181+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 64, in write_to_redis
[2025-04-25T11:52:56.182+0000] {subprocess.py:93} INFO -     if df.isEmpty(): return
[2025-04-25T11:52:56.182+0000] {subprocess.py:93} INFO -        ^^^^^^^^^^^^
[2025-04-25T11:52:56.183+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 553, in isEmpty
[2025-04-25T11:52:56.183+0000] {subprocess.py:93} INFO -     return self._jdf.isEmpty()
[2025-04-25T11:52:56.184+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:56.185+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-25T11:52:56.185+0000] {subprocess.py:93} INFO -     return_value = get_return_value(
[2025-04-25T11:52:56.185+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^
[2025-04-25T11:52:56.186+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 190, in deco
[2025-04-25T11:52:56.186+0000] {subprocess.py:93} INFO -     return f(*a, **kw)
[2025-04-25T11:52:56.187+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^
[2025-04-25T11:52:56.187+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-04-25T11:52:56.188+0000] {subprocess.py:93} INFO -     raise Py4JJavaError(
[2025-04-25T11:52:56.188+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o181.isEmpty.
[2025-04-25T11:52:56.189+0000] {subprocess.py:93} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 0.0 failed 1 times, most recent failure: Lost task 1.0 in stage 0.0 (TID 1) (237827c6637e executor driver): org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:56.189+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:56.190+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:56.190+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:56.191+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:56.191+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.191+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:56.192+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.193+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.193+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:56.194+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:56.194+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.194+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:56.196+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:56.196+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:56.197+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:56.197+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:56.198+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:56.198+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:56.199+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:56.199+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:56.200+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:56.200+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:56.201+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:56.201+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:56.202+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:56.202+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:56.202+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:56.203+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:56.203+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:56.204+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:56.204+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:56.205+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:56.205+0000] {subprocess.py:93} INFO - Driver stacktrace:
[2025-04-25T11:52:56.205+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2668)
[2025-04-25T11:52:56.206+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2604)
[2025-04-25T11:52:56.206+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2603)
[2025-04-25T11:52:56.206+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-04-25T11:52:56.206+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-04-25T11:52:56.207+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-04-25T11:52:56.207+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2603)
[2025-04-25T11:52:56.208+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1178)
[2025-04-25T11:52:56.208+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1178)
[2025-04-25T11:52:56.209+0000] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2025-04-25T11:52:56.209+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1178)
[2025-04-25T11:52:56.210+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2856)
[2025-04-25T11:52:56.210+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2798)
[2025-04-25T11:52:56.211+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2787)
[2025-04-25T11:52:56.211+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2025-04-25T11:52:56.212+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:952)
[2025-04-25T11:52:56.212+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2238)
[2025-04-25T11:52:56.213+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2259)
[2025-04-25T11:52:56.213+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2278)
[2025-04-25T11:52:56.213+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:506)
[2025-04-25T11:52:56.214+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:459)
[2025-04-25T11:52:56.214+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1(Dataset.scala:604)
[2025-04-25T11:52:56.215+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$isEmpty$1$adapted(Dataset.scala:603)
[2025-04-25T11:52:56.215+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:3898)
[2025-04-25T11:52:56.215+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:510)
[2025-04-25T11:52:56.216+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3896)
[2025-04-25T11:52:56.216+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:52:56.217+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:52:56.217+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:52:56.217+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:56.218+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:52:56.219+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3896)
[2025-04-25T11:52:56.219+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.Dataset.isEmpty(Dataset.scala:603)
[2025-04-25T11:52:56.220+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-04-25T11:52:56.220+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-04-25T11:52:56.221+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-04-25T11:52:56.221+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-04-25T11:52:56.221+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-04-25T11:52:56.222+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-04-25T11:52:56.222+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-04-25T11:52:56.223+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-04-25T11:52:56.223+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-04-25T11:52:56.224+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.sendCommand(ClientServerConnection.java:244)
[2025-04-25T11:52:56.224+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:384)
[2025-04-25T11:52:56.224+0000] {subprocess.py:93} INFO - 	at py4j.CallbackClient.sendCommand(CallbackClient.java:356)
[2025-04-25T11:52:56.225+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:106)
[2025-04-25T11:52:56.226+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-25T11:52:56.226+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-25T11:52:56.227+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-25T11:52:56.227+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-25T11:52:56.228+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-25T11:52:56.228+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-25T11:52:56.229+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-25T11:52:56.229+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-25T11:52:56.229+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:56.230+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-25T11:52:56.230+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-25T11:52:56.230+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:52:56.231+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:52:56.231+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:52:56.232+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-25T11:52:56.233+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-25T11:52:56.233+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:52:56.234+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-25T11:52:56.234+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-25T11:52:56.234+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-25T11:52:56.235+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-25T11:52:56.235+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-25T11:52:56.236+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-25T11:52:56.236+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-25T11:52:56.237+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-25T11:52:56.237+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-25T11:52:56.238+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-25T11:52:56.239+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-25T11:52:56.239+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Malformed records are detected in record parsing. Current parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
[2025-04-25T11:52:56.240+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:114)
[2025-04-25T11:52:56.240+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2025-04-25T11:52:56.241+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-25T11:52:56.241+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-25T11:52:56.241+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.242+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-25T11:52:56.242+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.243+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.243+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-25T11:52:56.244+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-25T11:52:56.244+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-25T11:52:56.245+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-25T11:52:56.245+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-25T11:52:56.245+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-25T11:52:56.246+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-25T11:52:56.246+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-25T11:52:56.247+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-25T11:52:56.248+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-25T11:52:56.249+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-25T11:52:56.250+0000] {subprocess.py:93} INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 29 out of bounds for length 2
[2025-04-25T11:52:56.250+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.parsing.Symbol$Alternative.getSymbol(Symbol.java:460)
[2025-04-25T11:52:56.251+0000] {subprocess.py:93} INFO - 	at org.apache.avro.io.ResolvingDecoder.readIndex(ResolvingDecoder.java:283)
[2025-04-25T11:52:56.252+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:188)
[2025-04-25T11:52:56.252+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:56.253+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readField(GenericDatumReader.java:260)
[2025-04-25T11:52:56.253+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readRecord(GenericDatumReader.java:248)
[2025-04-25T11:52:56.254+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.readWithoutConversion(GenericDatumReader.java:180)
[2025-04-25T11:52:56.254+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:161)
[2025-04-25T11:52:56.255+0000] {subprocess.py:93} INFO - 	at org.apache.avro.generic.GenericDatumReader.read(GenericDatumReader.java:154)
[2025-04-25T11:52:56.255+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.avro.AvroDataToCatalyst.nullSafeEval(AvroDataToCatalyst.scala:100)
[2025-04-25T11:52:56.256+0000] {subprocess.py:93} INFO - 	... 18 more
[2025-04-25T11:52:56.256+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:56.257+0000] {subprocess.py:93} INFO - 
[2025-04-25T11:52:56.383+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-04-25T11:52:56.399+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-04-25T11:52:56.406+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=traffic_data_pipeline, task_id=submit_traffic_processor_spark_job, execution_date=20250425T115239, start_date=20250425T115240, end_date=20250425T115256
[2025-04-25T11:52:56.425+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 16 for task submit_traffic_processor_spark_job (Bash command failed. The command returned a non-zero exit code 1.; 183)
[2025-04-25T11:52:56.469+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-25T11:52:56.518+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
